promtsdb/promtsdb.go

```diff
@@ -71,26 +85,32 @@ func (q *V3ioPromQuerier) Select(params *storage.SelectParams, oms ...*labels.Ma
 			}
 		}
 	}
-	set, err := q.q.SelectProm(name, functions, params.Step, filter, noAggr)
+	set, err := promQuery.v3ioQuerier.SelectProm(name, functions, params.Step, filter, noAggr)
 	return &V3ioPromSeriesSet{s: set}, err
 }
 
 // LabelValues returns all potential values for a label name.
-func (q *V3ioPromQuerier) LabelValues(name string) ([]string, error) {
-	return q.q.LabelValues(name)
+func (promQuery *V3ioPromQuerier) LabelValues(name string) ([]string, error) {
+	return promQuery.v3ioQuerier.LabelValues(name)
 }
 
 // Close releases the resources of the Querier.
-func (q *V3ioPromQuerier) Close() error {
+func (promQuery *V3ioPromQuerier) Close() error {
 	return nil
 }
 
-func match2filter(oms []*labels.Matcher) (string, string, string) {
-	filter := []string{}
+func match2filter(oms []*labels.Matcher, logger logger.Logger) (string, string, string) {
+	var filter []string
 	agg := ""
 	name := ""
 
+	if logger == nil {
+		// TODO: handle and return the error
+		logger, _ = utils.NewLogger(config.DefaultLogLevel)
+	}
+
 	for _, matcher := range oms {
+		logger.Debug("Matcher: %+v", matcher)
```

Comment:
Seems like it would be more useful to print if it actually matched something.

Reply:
it's not designed for unsupported matchers - it does not return no error